### PR TITLE
Upgrade datafusion to 20.0.0 & sqlparser to to 0.32.0

### PR DESCRIPTION
**Pull request by r4ntix**
# Which issue does this PR close?

Upgrade DataFusion to 20.0.0, sqlparser to 0.32.0

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by r4ntix** [comment]
TPC-H's q7 deserialization `proto.try_into_logical_plan(&ctx, codec.logical_extension_codec())` failed, I will analyze it again tomorrow.

**Comment by r4ntix** [comment]
The problem only occurs in debug build mode.
In datafusion, benchmarks-serde-related tests are used as a separate feature named `ci`, and the `release` build mode is used when running `ci` tests.
Maybe we need to do the same in ballista, @andygrove could you help with some suggestions?

**Comment by r4ntix** [comment]
@andygrove @Dandandan @yahoNanJing I have fixed the problem of test failure, could you help review it?

**Comment by yahoNanJing** [comment]
Thanks @r4ntix for upgrading the datafusion dependency. I'm good with the code and UT change. Let's wait for @andygrove to check whether it's OK for the workflow change.

**Review by yahoNanJing — APPROVED** [review]

File: ballista/client/Cargo.toml [hunk 1]
```diff
@@ -31,12 +31,12 @@ rust-version = "1.63"
 ballista-core = { path = "../core", version = "0.11.0" }
 ballista-executor = { path = "../executor", version = "0.11.0", optional = true }
 ballista-scheduler = { path = "../scheduler", version = "0.11.0", optional = true }
-datafusion = "19.0.0"
-datafusion-proto = "19.0.0"
+datafusion = { workspace = true }
+datafusion-proto = { workspace = true }
```
Comment (thinkharderdev) [inline ballista/client/Cargo.toml:35]: Nice :)

**Review by thinkharderdev — APPROVED** [review]

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @r4ntix

**Pull request merged**